server/src/server.ts

```diff
@@ -51,14 +51,15 @@ let currentActiveTextDocument: TextDocument = TextDocument.create(
 )
 
 connection.onInitialize(async (params: InitializeParams): Promise<InitializeResult> => {
-  logger.level = 'debug'
+  logger.level = params.initializationOptions?.loggingLevel ?? 'error'
   logger.info('[onInitialize] Initializing connection')
   bitBakeProjectScannerClient.setConnection(connection)
   disposables.push(...bitBakeProjectScannerClient.buildHandlers())
 
   workspaceFolder = params.workspaceFolders?.[0].uri.replace('file://', '')
 
-  const extensionPath = params.initializationOptions.extensionPath as string
+  const extensionPath = params.initializationOptions?.extensionPath ?? workspaceFolder
+  pokyFolder = extensionPath

```

Comment:
This mixes the extension sources domain and the workspace domain. I recommend to keep the existing pokyFolder assignment, but add undefined operators in the case where pathToBitbakeFolder is undefined like you did for other properties.

Reply:
Fixed